### PR TITLE
test(file-storage): real upload→store→download/delete e2e through CommandLayer

### PR DESCRIPTION
**Pull request by laofahai**
## The gap

cap-file-storage's existing tests cover the `LocalStorageAdapter` directly, but **nothing drove the `upload_file` / `download_file` / `delete_file` Actions**. The entire action layer was uncovered — base64 decode/encode, input validation, server-side **tenant-prefixed key generation**, the **orphan-cleanup compensation** path, the adapter-name guard, and `file.uploaded` / `file.deleted` event emission. An action-level wiring bug would ship undetected. (Same fake-coverage theme as the auth/permission/mcp/notification e2e PRs this round.)

## What this adds

`addons/file-storage/cap-file-storage/__tests__/upload-download-e2e.test.ts` drives the REAL pipeline via `commandLayer.execute({ command: "upload_file" | "download_file" | "delete_file", ... })` (full auth→exposure→permission→tenant→action slots) with a **real `LocalStorageAdapter`** on a temp dir and a **real `EventBus`** — nothing mocked. DB-free and hermetic (no S3/cloud creds, no external service).

5 tests:
- byte + metadata round-trip (asserts size, checksum, adapter, `uploaded_by`, a tenant-prefixed storage key that does NOT echo the filename, and the `file.uploaded` event)
- UTF-8 text round-trip
- delete-then-download-fails (+ `file.deleted` event)
- idempotent delete of a missing id
- pipeline rejects a missing required field

`afterAll` cleans the temp dir, guarded by `if (root)` so a setup failure isn't masked.

## Verification

- `bun test <file>` → 5 pass (28 in the package).
- `bun run check` (real biome 2.4.16) clean; `bun run typecheck` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive end-to-end test suite for file storage operations including upload, download, and delete functionality.
  * Tests cover successful file operations, edge cases (missing files, invalid uploads), and event emission validation for various content types.

### CODE REVIEW

File: addons/file-storage/cap-file-storage/__tests__/upload-download-e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    expect(typeof fileId).toBe("string");
+    expect(record.name).toBe("blob.bin");
+    expect(record.mime).toBe("application/octet-stream");
+    expect(record.size).toBe(payload.byteLength);
+    expect(record.adapter).toBe("local");
+    expect(record.checksum).toBe(expectedChecksum);
+    expect(record.uploaded_by).toBe(ACTOR.id);
+    // Server-generated key MUST be scoped under the tenant prefix and MUST NOT
+    // echo the original filename (anti cross-tenant-overwrite invariant).
+    expect(record.path as string).toStartWith(`${TENANT}/`);
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/file-storage/cap-file-storage/__tests__/upload-download-e2e.test.ts]: Switched to the portable form: `expect((record.path as string).startsWith(`${TENANT}/`)).toBe(true)`. (Note: `.toStartWith` is in fact a valid Bun matcher — the suite passed with it — but the standard `.startsWith()` form is unambiguous and tool-agnostic, so it's the better choice here.)

File: addons/file-storage/cap-file-storage/__tests__/upload-download-e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    // Register the three real action definitions on the real registry.
+    for (const action of [
+      uploadFileAction,
+      downloadFileAction,
+      deleteFileAction,
+    ] as ActionDefinition[]) {
+      executor.registry.register(action);
+    }
+    commandLayer = createCommandLayer({ executor });
+  });
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/file-storage/cap-file-storage/__tests__/upload-download-e2e.test.ts]: Fixed — moved the EventBus + InMemoryStore + executor + CommandLayer creation (and the subscriptions) into beforeEach, so every test gets a fresh runtime. This gives full isolation and makes subscription leaks impossible, so the afterEach `eventBus.clear()` is gone. The shared temp dir / storage adapter stays in beforeAll (tests use unique ids).
